crates/turborepo-lib/src/commands/mod.rs

```diff
@@ -1,2 +1,40 @@
+use std::path::PathBuf;
+
+use anyhow::Result;
+
+use crate::{
+    config::{
+        default_user_config_path, RepoConfig, RepoConfigLoader, UserConfig, UserConfigLoader,
+    },
+    ui::UI,
+    Args,
+};
+
 pub(crate) mod bin;
 pub(crate) mod logout;
+
+pub struct CommandBase {
+    pub repo_root: PathBuf,
+    pub ui: UI,
+    pub repo_config: RepoConfig,
+    pub user_config: UserConfig,
```

Comment:
Not blocking, but it would be nice if we were lazy about reading these configs to avoid unnecessary fs operations.

Reply:
We could have these be methods and store the args in the struct. Would be a little more copying but not a big deal